franklin/blocks/page-feed/page-feed.js

```diff
@@ -0,0 +1,383 @@
+import {
+  createTag,
+  getConfig,
+  replaceKey,
+} from '../../scripts/scripts/scripts.js';
+
+function turnH6intoDetailM(scope) {
+  scope.querySelectorAll('h6').forEach((h6) => {
+    const p = createTag('p', { class: 'detail-m' }, h6.innerHTML);
+    const attrs = h6.attributes;
+    for (let i = 0, len = attrs.length; i < len; i += 1) {
+      p.setAttribute(attrs[i].name, attrs[i].value);
+    }
+    h6.parentNode.replaceChild(p, h6);
+  });
+}
+
+function toSentenceCase(str) {
+  return (str && typeof str === 'string') ? str.toLowerCase().replace(/(^\s*\w|[.!?]\s*\w)/g, (c) => c.toUpperCase()) : '';
+}
+
+function transformLinkToAnimation(a) {
+  if (!a || !a.href.includes('.mp4')) return null;
+  const params = new URL(a.href).searchParams;
+  const attribs = {};
+  ['playsinline', 'autoplay', 'loop', 'muted'].forEach((p) => {
+    if (params.get(p) !== 'false') attribs[p] = '';
+  });
+  // use the closest picture as poster
+  const poster = a.closest('div').querySelector('picture source');
+  if (poster) {
+    attribs.poster = poster.srcset;
+    poster.parentNode.remove();
+  }
+  // replace anchor with video element
+  const videoUrl = new URL(a.href);
+  const video = createTag('video', attribs);
+  video.innerHTML = `<source src="${videoUrl}" type="video/mp4">`;
+  const innerDiv = a.closest('div');
+  innerDiv.prepend(video);
+  innerDiv.classList.add('hero-animation-overlay');
+  a.replaceWith(video);
+  // autoplay animation
+  video.addEventListener('canplay', () => {
+    video.muted = true;
+    video.play();
+  });
+  return video;
+}
+
+export function makeRelative(href) {
+  const projectName = 'adobestock--adobecom';
+  const productionDomains = ['stock.adobe.com'];
+  const fixedHref = href.replace(/\u2013|\u2014/g, '--');
+  const hosts = [`${projectName}.hlx.page`, `${projectName}.hlx.live`, ...productionDomains];
+  const url = new URL(fixedHref);
+  const relative = hosts.some((host) => url.hostname.includes(host))
+    || url.hostname === window.location.hostname;
+  return relative ? `${url.pathname}${url.search}${url.hash}` : href;
+}
+
+function getFetchRange(payload) {
+  let range;
+
+  if (payload.offset + payload.limit < payload.cardsToBuild.length) {
+    range = payload.offset + payload.limit;
+  } else {
+    range = payload.cardsToBuild.length;
+  }
+
+  return range;
+}
+
+export async function loadPageFeedCard(a) {
+  if (!a) return null;
+
+  const href = (typeof (a) === 'string') ? a : a.href;
+  const path = makeRelative(href);
+  if (!path.startsWith('/')) return null;
```

Comment:
Is this to prevent arbitrary fetches? Did you ensure code cannot be escaped to run arbitrary fetches in the client?

Reply:
This is also a migrated function. It was in the utils.js in the old repo and was shared by multiple blocks. I extracted to have it only work for this block.
Just reading the code, I believe what it's trying to do is to do is to only make the attempt to fetch and load the corresponding page-feed-card when the path is an internal link.
So the escape logic is:
- if a link is not an artisthub link, don't fetch and load the page-feed-card
  - if a link is an artisthub link, but fetch didn't return a page-feed-card, throw a false-y value, and at card render, the card index is logged in 'trash can' and pagination is skipped.

This is actually one of the reasons why the page-feed block remains valuable in the migration -- it's versatile enough to support links to any source of content. Be that an external link, a pdf file or a modal link.
That said, the makeRelative function naming is surely a little confusing. I'll change that.